### DISCUSSION TITLE
Possible to update MoltenVK?  (...and the difference between the wrapper and engine?)

### DISCUSSION THREAD

**Discussion by sofakng** [The-Wineskin-Project/WineskinServer#373] [General]
Is it possible to update MoltenVK separately from the other components?

It looks like MoltenVK (ie. libMoltenVK.dylib) is part of the **wrapper** and not the **engine**, is that correct?  Therefore all the engines use the same version of MoltenVK?

Here is the only location I've find the MoltenVK library:
`~/Library/Application Support/Wineskin/Wrapper/Wineskin-2.9.1.8.app/Contents/Frameworks/libMoltenVK.dylib
`

I'm also confused because [Gcenx/winecx](https://github.com/Gcenx/winecx/releases) releases such as **crossover-wine-22.0.1** reference including **MoltenVK @1.2.0_0** but I thought winecx is an engine?

Also, can somebody explain the relationship between the wrapper and engine?

**Comment by Gcenx** [comment]
> Is it possible to update MoltenVK separately from the other components?

Yes, you’d do this manually.

> It looks like MoltenVK (ie. libMoltenVK.dylib) is part of the **wrapper** and not the **engine**, is that correct? Therefore all the engines use the same version of MoltenVK?

Yep, the master wrapper contains MoltenVK along with all the other required dylibs engines topically won’t contain dylibs.

> Here is the only location I've find the MoltenVK library: `~/Library/Application Support/Wineskin/Wrapper/Wineskin-2.9.1.8.app/Contents/Frameworks/libMoltenVK.dylib `

That’s the master wrapper, that’s the base of any generated wrapper you can change the copy of `libMoltenVK.dylib` within your generated wrappers.

> I'm also confused because [Gcenx/winecx](https://github.com/Gcenx/winecx/releases) releases such as **crossover-wine-22.0.1** reference including **MoltenVK @1.2.0_0** but I thought winecx is an engine?

As explained at the very top of that repo
> Codeweavers modified wine version. This repository is just a mirror of https://media.codeweavers.com/pub/crossover/source/

The packages provided there are not the same as the “WineCX” engines here, those packages are completely self contained Winehq like packages so need I need to bundle all required dylibs within the app bundle.

> Also, can somebody explain the relationship between the wrapper and engine?

The “master wrapper” contains all needed resources to run wine. \
This includes;
- WineskinLauncher (the actual launcher)
- Wineskin (the configuration UI)
- “Runtime” (dylibs)

What’s an Engine?\
An Engine is a compiled and packaged copy of “wine” (with its corisponding gecko/mini version)

Together these become the basis of your wrapper or “port”

**Reply by sofakng** [reply]
Thank you so much for the information and I apologize for all of the questions!

Is it possible to determine the version of the libMoltenVK.dylib from the file itself?

For example, otool only displays "version 1.0.0" for libMoltenVK.dylib which doesn't seem to match anything from [your MoltenVK repo](https://github.com/The-Wineskin-Project/MoltenVK/releases).

Do I need to reference [your wiki](https://github.com/Gcenx/WineskinServer/wiki) to see which version is included?

Thanks again for all of your amazing and hard work on this project!

**Reply by Gcenx** [reply]
You’d check the wiki for the current MoltenVK version (1.2.2)

You’d need to ask upstream/MoltenVK about the version string, but usually the reason would be if a library reports 1.0 even after many versions it means it’s still backwards compatible with prior releases.

**Reply by sofakng** [reply]
Thank you so much for the help!
